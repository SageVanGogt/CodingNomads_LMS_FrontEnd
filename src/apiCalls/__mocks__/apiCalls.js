import { mockDocs } from './../../mockData/mockDocs';
import { mockLabs } from './../../mockData/mockLabs';
import { mockTasks } from './../../mockData/mockTasks';
import { mockCourses } from './../../mockData/mockCourses';

export const getAllTasks = jest.fn().mockImplementation(
  () => Promise.resolve(mockTasks)
);

export const getTask = jest.fn();

export const postTask = jest.fn();

export const updateTask = jest.fn();

export const deleteTask = jest.fn();

export const getAllCourses = jest.fn().mockImplementation(
  () => Promise.resolve(mockCourses)
);

export const getCourse = jest.fn();

export const addCourse = jest.fn();

export const updateCourse = jest.fn();

export const deleteCourse = jest.fn();

export const getAllCoursesStudents = jest.fn();

export const addCourseStudent = jest.fn();

export const deleteCourseStudent = jest.fn();

export const getCourseTeachers = jest.fn();

export const addTeacherToCourse = jest.fn();

export const deleteTeacherFromCourse = jest.fn();

export const getCourseTasks = jest.fn();

export const postTaskToCourse = jest.fn();

export const deleteTaskFromCourse = jest.fn();

export const getAllDocs = jest.fn().mockImplementation(
  () => Promise.resolve(mockDocs)
);

export const getAllLabs = jest.fn().mockImplementation(
  () => Promise.resolve(mockLabs)
);

export const deleteLabsFromTask = jest.fn(); 

export const deleteDocsFromTask = jest.fn();

export const deleteTasksFromCourse = jest.fn();

export const deleteStudentsFromCourse = jest.fn();



