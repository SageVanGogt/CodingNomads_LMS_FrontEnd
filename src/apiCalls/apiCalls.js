import { apiPath } from '../constants/apiPath';

export const getAllTasks = async () => {
  const url = apiPath + '/tasks';
  try {
    const response = await fetch(url);
    if (response.status !== 200) {
      throw Error('Could not fetch tasks');
    } else {
      return await response.json();
    }
  } catch (error) {
    throw error;
  }
};

// export const getAllTasksAuthed = async () => {
//   const token = localStorage.getItem('access_token') || null;
//   const url = apiPath + '/tasks';
//   const options = {
//     headers: { 'Authorization': `Bearer ${token}` }
//   }
//   try {
//     const response = await fetch(url);
//     if (response.status !== 200) {
//       throw Error('Could not fetch tasks');
//     } else {
//       return await response.json();
//     }
//   } catch (error) {
//     throw error;
//   }
// };

export const getTask = async (taskId) => {
  const url = apiPath + `/tasks/${taskId}`;
  try {
    const response = await fetch(url);
    if (response.status !== 200) {
      throw Error('Task could not be found.');
    } else {
      return await response.json();
    }
  } catch (error) {
    throw error;
  }
};

export const postTask = async (newTask) => {
  const url = apiPath + '/tasks';
  const options = {
    method: 'POST',
    headers: { 'content-type': 'application/json' },
    body: JSON.stringify(newTask)
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 201) {
      throw Error('Could not post task');
    } else {
      return await response.json();
    }
  } catch (error) {
    throw error;
  }
}; 

export const updateTask = async (updatedTask) => {
  const url = apiPath + `/tasks/${updatedTask.id}`;
  const options = {
    method: 'PATCH',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(updatedTask)
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 200) {
      throw Error('Could not patch request');
    } else {
      return await response.json();
    }
  } catch (error) {
    throw error;
  }
};

export const deleteTask = async (taskId) => {
  const url = apiPath + `/tasks/${taskId}`;
  const options = {
    method: 'DELETE',
    headers: { 'content-type': 'application/json' }
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 204) {
      throw Error('That id could not be found.');
    }
  } catch (error) {
    throw error;
  }
};

export const getAllCourses = async () => {
  const url = apiPath + '/courses';
  try {
    const response = await fetch(url);
    if (response.status !== 200) {
      throw Error('Could not get courses');
    } else {
      return await response.json();
    }
  } catch (error) {
    throw error;
  }
};

export const getCourse = async (courseId) => {
  const url = apiPath + `/courses/${courseId}`;
  try {
    const response = await fetch(url);
    if (response.status !== 200) {
      throw Error('Could not find course');
    } else {
      return await response.json();
    }
  } catch (error) {
    throw error;
  }
};

export const addCourse = async (newCourse) => {
  const url = apiPath + '/courses';
  const options = {
    method: 'POST',
    headers: { 'content-type': 'application/json' },
    body: JSON.stringify(newCourse)
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 200) {
      throw Error('Could not add course');
    } else {
      return await response.json();
    }
  } catch (error) {
    throw error;
  }
}; 

export const updateCourse = async (updatedCourse) => {
  const url = apiPath + `/courses`;
  const options = {
    method: 'PATCH',
    headers: { 'content-type': 'application/json' },
    body: JSON.stringify(updatedCourse)
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 200) {
      throw Error('Failed to update course.');
    } else {
      return await response.json();
    }
  } catch (error) {
    throw error;
  }
};

export const deleteCourse = async (courseId) => {
  const url = apiPath + `/courses/${courseId}`;
  const options = {
    method: 'DELETE',
    headers: { 'content-type': 'application/json' }
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 204) {
      throw Error('That id could not be found.');
    }
  } catch (error) {
    throw error;
  }
};

export const getAllCourseStudents = async (courseId) => {
  const url = apiPath + `/courses/${courseId}/students`;
  try {
    const response = await fetch(url);
    if (response.status !== 200) {
      throw Error('Could not get students in course');
    } else {
      return await response.json();
    }
  } catch (error) {
    throw error;
  }
};

export const addCourseStudent = async (newCourseStudent, course) => {
  const studentCourseRelation = {
    studentId: newCourseStudent.id,
    courseId: course.id
  };
  const url = apiPath + '/courses/students';
  const options = {
    method: 'POST',
    headers: { 'content-type': 'application/json' },
    body: JSON.stringify(studentCourseRelation)
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 201) {
      throw Error('Could not add student to course');
    } else {
      return await response.json();
    }
  } catch (error) {
    throw error;
  }
}; 

export const deleteCourseStudent = async (student, course) => {
  const url = apiPath + `/courses/${course.id}/student/${student.id}`;
  const options = {
    method: 'DELETE',
    headers: { 'content-type': 'application/json' }
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 204) {
      throw Error('That id could not be found.');
    }
  } catch (error) {
    throw error;
  }
};

export const getCourseTeachers = async (courseID) => {
  const url = apiPath + `/courses/${courseID}/teachers`;
  try {
    const response = await fetch(url);
    if (response.status !== 200) {
      throw Error('Could not get teachers');
    } else {
      return await response.json();
    }
  } catch (error) {
    throw error;
  }
};

export const addTeacherToCourse = async (courseId, teacherID) => {
  const url = apiPath + `/courses/${courseId}/teachers`;
  const options = {
    method: 'POST',
    headers: { 'content-type': 'application/json' },
    body: JSON.stringify(teacherID)
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 201) {
      throw Error('Could not add teacher to course');
    } else {
      return await response.json();
    }
  } catch (error) {
    throw error;
  }
};

export const deleteTeacherFromCourse = async (courseId, teacherId) => {
  const url = apiPath + `/courses/${courseId}/teachers/${teacherId}`;
  const options = {
    method: 'DELETE',
    headers: { 'content-type': 'application/json' }
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 204) {
      throw Error('That id could not be found.');
    }
  } catch (error) {
    throw error;
  }
};

export const getCourseTasks = async (courseID) => {
  const url = apiPath + `/courses/${courseID}/tasks`;
  try {
    const response = await fetch(url);
    if (response.status !== 200) {
      throw Error('Could not get tasks for that course');
    } else {
      return await response.json();
    }
  } catch (error) {
    throw error;
  }
};

export const postTaskToCourse = async (courseId, taskID) => {
  const url = apiPath + `/courses/${courseId}/tasks`;
  const options = {
    method: 'POST',
    headers: { 'content-type': 'application/json' },
    body: JSON.stringify(taskID)
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 201) {
      throw Error('Could not add task to course');
    } else {
      return await response.json();
    }
  } catch (error) {
    throw error;
  }
};

export const deleteTaskFromCourse = async (courseId, taskID) => {
  const url = apiPath + `/courses/${courseId}/tasks/${taskID}`;
  const options = {
    method: 'DELETE',
    headers: { 'content-type': 'application/json' }
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 204) {
      throw Error('That id could not be found.');
    }
  } catch (error) {
    throw error;
  }
};

export const getAllDocs = async () => {
  const url = apiPath + '/docs';
  try {
    const response = await fetch(url);
    if (response.status !== 200) {
      throw Error('');
    } else {
      return await response.json();
    }
  } catch (error) {
    throw error;
  }
};

export const getAllLabs = async () => {
  const url = apiPath + '/labs';
  try {
    const response = await fetch(url);
    if (response.status !== 200) {
      throw Error('');
    } else {
      return await response.json();
    }
  } catch (error) {
    throw error;
  }
};

export const deleteDocsFromTask = async (taskId, docsArr) => {
  const url = apiPath + `/task/${taskId}/docs`;
  const options = {
    method: 'DELETE',
    headers: { 'content-type': 'application/json' },
    body: JSON.stringify({ docs: docsArr })
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 204) {
      throw Error('That id could not be found.');
    }
  } catch (error) {
    throw error;
  }
};

export const deleteLabsFromTask = async (taskId, labsArr) => {
  const url = apiPath + `/task/${taskId}/labs`;
  const options = {
    method: 'DELETE',
    headers: { 'content-type': 'application/json' },
    body: JSON.stringify({ labs: labsArr })
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 204) {
      throw Error('That id could not be found.');
    }
  } catch (error) {
    throw error;
  }
};

export const deleteTasksFromCourse = async (courseId, tasksArr) => {
  const url = apiPath + `/course/${courseId}/tasks`;
  const options = {
    method: 'DELETE',
    headers: { 'content-type': 'application/json' },
    body: JSON.stringify({ tasks: tasksArr })
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 204) {
      throw Error('That id could not be found.');
    }
  } catch (error) {
    throw error;
  }
};

export const deleteStudentsFromCourse = async (courseId, studentsArr) => {
  const url = apiPath + `/course/${courseId}/students`;
  const options = {
    method: 'DELETE',
    headers: { 'content-type': 'application/json' },
    body: JSON.stringify({ students: studentsArr })
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 204) {
      throw Error('That id could not be found.');
    }
  } catch (error) {
    throw error;
  }
};

export const authUser = async (user) => {
  const url = apiPath + '/users'; 
  const options = {
    method: 'POST',
    headers: { 'content-type': 'application/json' },
    body: JSON.stringify(user)
  };
  try {
    const response = await fetch(url, options);
    if (response.message !== 'success') {
      throw Error('User does not exist');
    } 
    const userAuth = await response.json();
    return userAuth;
  } catch (error) {
    throw error;
  }
};


