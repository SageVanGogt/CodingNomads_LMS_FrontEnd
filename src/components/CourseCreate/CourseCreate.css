.CourseCreate_page {
  margin-left: 200px;
  width: calc(100% - 200px);
  padding-top: 20px;
  background-color: #efefef; }

.CourseCreate_page h1 {
  font-size: 36px;
  margin-bottom: 20px;
  color: #666666; }

.CourseCreate_page h2 {
  font-size: 20px;
  margin-top: 20px;
  color: #666666; }

.CourseCreate_form input {
  margin-top: 20px;
  height: 40px;
  width: 400px;
  padding-left: 15px;
  background-color: transparent;
  border: none;
  border-bottom: 3px solid #4885a2;
  font-size: 1.2rem;
  font-family: 'Open Sans', sans-serif;
  display: block;
  margin: 10px auto; }

.CourseCreate_form textarea {
  display: block;
  margin: 10px auto;
  width: 400px;
  min-height: 100px;
  border: none;
  border-bottom: 3px solid #4885a2; }

.CourseCreate_form button {
  height: 40px;
  margin-top: 15px;
  border: none;
  background-color: #0F94FB;
  font-size: 1.2rem;
  cursor: pointer;
  color: #efefef;
  font-weight: 600;
  -webkit-box-shadow: 0 5px 6px -4px rgba(0, 0, 0, 0.22), 0 10px 13px -4px rgba(0, 0, 0, 0.34), inset 0 30px 30px 0 rgba(255, 255, 255, 0.35), inset 0 -30px 30px 0 rgba(0, 0, 0, 0.1);
  box-shadow: 0 5px 6px -4px rgba(0, 0, 0, 0.22), 0 10px 13px -4px rgba(0, 0, 0, 0.34), inset 0 30px 30px 0 rgba(255, 255, 255, 0.35), inset 0 -30px 30px 0 rgba(0, 0, 0, 0.1); }

.tasksArea {
  width: 400px;
  min-height: 200px;
  border: 2px solid #0F94FB;
  margin: 20px auto; }

.studentsArea {
  width: 400px;
  height: 150px;
  border: 2px solid #0F94FB;
  margin: 20px auto;
  overflow: scroll; }

.CourseStudentCard {
  width: 300px;
  text-decoration: underline;
  text-decoration-color: #0F94FB;
  margin: 5px auto; }

.course_message {
  padding-bottom: 20px; }
