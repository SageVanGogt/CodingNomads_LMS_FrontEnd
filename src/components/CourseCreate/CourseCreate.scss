.CourseCreate_page {
  margin-left: 200px;
  width: calc(100% - 200px);
  padding-top: 20px;
}

.CourseCreate_page h1 {
  font-size: 36px;
  margin-bottom: 20px;
}

.CourseCreate_page h2 {
  font-size: 20px;
  margin-top: 20px;
}

.CourseCreate_form input {
  display: block;
  margin: 10px auto;
  width: 400px;
  height: 20px;
}

.CourseCreate_form textarea {
  display: block;
  margin: 10px auto;
  width: 400px;
  min-height: 100px;
}

.CourseCreate_form button {
  display: block;
  margin: 20px auto
}

.tasksArea {
  width: 400px;
  min-height: 200px;
  border: .5px solid;
  margin: 20px auto;
}

.studentsArea {
  width: 400px;
  height: 150px;
  border: .5px solid;
  margin: 20px auto;
  overflow: scroll;
}

.CourseStudentCard {
  width: 300px;
  border: 1px dashed black;
  margin: 5px auto;
}