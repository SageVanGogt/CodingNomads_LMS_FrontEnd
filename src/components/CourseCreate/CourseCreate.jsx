import React, { Component } from 'react';
import { connect } from 'react-redux';
import { mockTasks } from '../../mockData/mockTasks';
import './CourseCreate.css';
import PropTypes from 'prop-types';
import * as apiCalls from '../../apiCalls/apiCalls';
// import { CourseTaskCard } from '../CourseTaskCard/CourseTaskCard';

import { CourseTaskContainer } from '../CourseTasksContainer/CourseTaskContainer';

export class CourseCreate extends Component {
  constructor(props) {
    super(props);
    
    this.state = {
      name: this.props.currentCourse.name,
      description: this.props.currentCourse.description,
      tasks: this.props.currentCourse.tasks || [],
      students: this.props.currentCourse.students || [],
      tasksToDelete: [],
      labsToDelete: [],
      allTasks: [],
      allStudents: []
    };
  }

  handleChange = (event) => {
    event.preventDefault();
    const { name, value } = event.target;

    this.setState({
      [name]: value
    });
  }

  rearrangeTasks = (tasks) => {
    this.setState({ tasks });
  }

  fetchTasks = async () => {
    const response = await fetch('https://cors-anywhere.herokuapp.com/54.191.130.113:8080/api/admin/v1/tasks');
    const tasks = await response.json();
    this.setState({allTasks: [{name: 'Select Task'}, ...tasks.data]});
  }

  fetchStudents = async () => {
    const response = await fetch('https://cors-anywhere.herokuapp.com/54.191.130.113:8080/api/admin/v1/students');
    const students = await response.json();
    this.setState({ allStudents: [{ firstName: 'Select', lastName: 'Student' }, ...students.data] });
  }
  
  handleTaskSelect = (e) => {
    const task = this.state.allTasks.find(task => task.name === e.target.value);
    if (!this.state.tasks.includes(task) && task.name !== "Select Task") {
      this.setState({
        tasks: [...this.state.tasks, task]
      });
    }
  }

  handleStudentSelect = (e) => {
    const student = this.state.allStudents.find(student => student.name === e.target.value);
    if (!this.state.students.includes(student) && student.name !== "Select Student") {
      this.setState({
        students: [...this.state.students, student]
      });
    }
  }

  componentDidMount() {
    this.fetchTasks();
    this.fetchStudents();
  }

  render() {
    const tasks = this.state.allTasks.map((task, index) => {
      return (
        <option
          key={`task-${index}`}
          name="task"
          title={task.description}>
          {task.name}
        </option>
      );
    });

    const students = this.state.allStudents.map((student, index) => {
      return (
        <option
          key={`student-${index}`}
          name="student">
          {student.firstName + ' ' + student.lastName}
        </option>
      );
    });

    return (
      <div className="CourseCreate_page">
        <h1>Edit course</h1>
        <form 
          action="submit" 
          className="CourseCreate_form"
          onSubmit={(e) => e.preventDefault()}
        >
          <input
            type="text"
            placeholder="Name"
            name="name"
            onChange={this.handleChange}
            value={this.state.name}
          />
          <textarea
            type="text"
            placeholder="Description"
            name="description"
            onChange={this.handleChange}
            value={this.state.description}
          />
          <h2>Tasks</h2>
          <CourseTaskContainer tasks={this.state.tasks} rearrangeTasks={this.rearrangeTasks}/>
          <span>Add task: </span>
          <select
            onChange={(e) => this.handleTaskSelect(e)}
          >
            {tasks}
          </select>

          <select
            onChange={(e) => this.handleStudentSelect(e)}
          >
            {students}
          </select>
          <button type="submit">Submit Course</button>
        </form>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  currentCourse: state.currentCourse
});

export default connect(mapStateToProps, null)(CourseCreate);