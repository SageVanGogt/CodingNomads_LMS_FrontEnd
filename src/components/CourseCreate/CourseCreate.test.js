import React from 'react';
import { shallow } from 'enzyme';
import { CourseCreate, mapStateToProps } from './CourseCreate';
import * as API from './../../apiCalls/apiCalls';

jest.mock('./../../apiCalls/apiCalls');

describe('CourseCreate', () => {
  let wrapper;
  let mockCurrentCourse;
  let mockProps;
  let mockTask;
  let fetch;

  beforeEach(() => {
    mockProps = {
      history: { push: jest.fn() },
      currentCourse: mockCurrentCourse
    };
    mockCurrentCourse = { 
      id: 1, 
      name: '',
      description: '',
      location: {},
      students: [],
      tasks: []
    };

    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      status: 500,
      json: jest.fn().mockImplementation(() => Promise.resolve({ data: [] }))
    }));

    wrapper = shallow(<CourseCreate {...mockProps} />, {disableLifeCycleMethods: true});
  })

  it('should match the snapshot', () => {
    
  });

  describe('handleChange', () => {
    it('should update the appropriate state', () => {
      let mockEvent = {
        preventDefault: jest.fn(),
        target: {
          name: 'name',
          value: 'javascript'
        }
      };
      let expected = 'javascript';
      wrapper.instance().handleChange(mockEvent);
      let actual = wrapper.state('name');

      expect(actual).toEqual(expected);
    });
  });

  describe('handleStudentSelect', () => {
    it('should call  with the correct params', async () => {
      let mockLabs = [{}, {}];
      wrapper.setState({ labsToDelete: mockLabs });
      let expected = mockLabs;
      await wrapper.instance().handleDeletedLabs();

      expect(API.deleteLabsFromTask).toHaveBeenCalledWith(expected);
    })
  })

  // describe('handleDeletedDocs', () => {
  //   it('should call  with the correct params', async () => {
  //     let mockDocs = [{}, {}];
  //     wrapper.setState({ docsToDelete: mockDocs });
  //     let expected = mockDocs;
  //     await wrapper.instance().handleDeletedDocs();

  //     expect(API.deleteDocsFromTask).toHaveBeenCalledWith(expected);
  //   })
  // })

  // describe('mapStateToProps', () => {
  //   it('should return a state with the currentTask prop', () => {
  //     let mockState = {
  //       currentTask: { id: null, name: '' },
  //       user: { id: 1 }
  //     };
  //     let mappedProps = mapStateToProps(mockState);
  //     let actual = mappedProps.currentTask;
  //     let expected = { id: null, name: '' };

  //     expect(actual).toEqual(expected);
  //   })
  // })
})
