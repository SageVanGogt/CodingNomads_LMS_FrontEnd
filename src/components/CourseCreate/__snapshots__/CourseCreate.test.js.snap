// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseCreate should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CourseCreate
    currentCourse={
        Object {
            "description": "",
            "id": 1,
            "name": "",
            "students": Array [],
            "tasks": Array [],
          }
    }
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Edit course
</h1>,
        <form
          className="CourseCreate_form"
>
          <input
                    name="name"
                    onChange={[Function]}
                    placeholder="Name"
                    type="text"
                    value=""
          />
          <textarea
                    name="description"
                    onChange={[Function]}
                    placeholder="Description"
                    type="text"
                    value=""
          />
          <h2>
                    Tasks
          </h2>
          <CourseTaskContainer
                    deleteTask={[Function]}
                    rearrangeTasks={[Function]}
                    tasks={Array []}
          />
          <span>
                    Add task: 
          </span>
          <select
                    onChange={[Function]}
          >
                    
          </select>
          <h2>
                    Students
          </h2>
          <div
                    className="studentsArea"
          >
                    
          </div>
          <span>
                    Add task: 
          </span>
          <select
                    onChange={[Function]}
          >
                    
          </select>
          <button
                    onClick={[Function]}
                    type="submit"
          >
                    Submit Course
          </button>
</form>,
      ],
      "className": "CourseCreate_page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Edit course",
        },
        "ref": null,
        "rendered": "Edit course",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="name"
              onChange={[Function]}
              placeholder="Name"
              type="text"
              value=""
/>,
            <textarea
              name="description"
              onChange={[Function]}
              placeholder="Description"
              type="text"
              value=""
/>,
            <h2>
              Tasks
</h2>,
            <CourseTaskContainer
              deleteTask={[Function]}
              rearrangeTasks={[Function]}
              tasks={Array []}
/>,
            <span>
              Add task: 
</span>,
            <select
              onChange={[Function]}
>
              
</select>,
            <h2>
              Students
</h2>,
            <div
              className="studentsArea"
>
              
</div>,
            <span>
              Add task: 
</span>,
            <select
              onChange={[Function]}
>
              
</select>,
            <button
              onClick={[Function]}
              type="submit"
>
              Submit Course
</button>,
          ],
          "className": "CourseCreate_form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "name",
              "onChange": [Function],
              "placeholder": "Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "description",
              "onChange": [Function],
              "placeholder": "Description",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tasks",
            },
            "ref": null,
            "rendered": "Tasks",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "deleteTask": [Function],
              "rearrangeTasks": [Function],
              "tasks": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add task: ",
            },
            "ref": null,
            "rendered": "Add task: ",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Students",
            },
            "ref": null,
            "rendered": "Students",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "studentsArea",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add task: ",
            },
            "ref": null,
            "rendered": "Add task: ",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit Course",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit Course",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Edit course
</h1>,
          <form
            className="CourseCreate_form"
>
            <input
                        name="name"
                        onChange={[Function]}
                        placeholder="Name"
                        type="text"
                        value=""
            />
            <textarea
                        name="description"
                        onChange={[Function]}
                        placeholder="Description"
                        type="text"
                        value=""
            />
            <h2>
                        Tasks
            </h2>
            <CourseTaskContainer
                        deleteTask={[Function]}
                        rearrangeTasks={[Function]}
                        tasks={Array []}
            />
            <span>
                        Add task: 
            </span>
            <select
                        onChange={[Function]}
            >
                        
            </select>
            <h2>
                        Students
            </h2>
            <div
                        className="studentsArea"
            >
                        
            </div>
            <span>
                        Add task: 
            </span>
            <select
                        onChange={[Function]}
            >
                        
            </select>
            <button
                        onClick={[Function]}
                        type="submit"
            >
                        Submit Course
            </button>
</form>,
        ],
        "className": "CourseCreate_page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Edit course",
          },
          "ref": null,
          "rendered": "Edit course",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="name"
                onChange={[Function]}
                placeholder="Name"
                type="text"
                value=""
/>,
              <textarea
                name="description"
                onChange={[Function]}
                placeholder="Description"
                type="text"
                value=""
/>,
              <h2>
                Tasks
</h2>,
              <CourseTaskContainer
                deleteTask={[Function]}
                rearrangeTasks={[Function]}
                tasks={Array []}
/>,
              <span>
                Add task: 
</span>,
              <select
                onChange={[Function]}
>
                
</select>,
              <h2>
                Students
</h2>,
              <div
                className="studentsArea"
>
                
</div>,
              <span>
                Add task: 
</span>,
              <select
                onChange={[Function]}
>
                
</select>,
              <button
                onClick={[Function]}
                type="submit"
>
                Submit Course
</button>,
            ],
            "className": "CourseCreate_form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "name",
                "onChange": [Function],
                "placeholder": "Name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "description",
                "onChange": [Function],
                "placeholder": "Description",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tasks",
              },
              "ref": null,
              "rendered": "Tasks",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "deleteTask": [Function],
                "rearrangeTasks": [Function],
                "tasks": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add task: ",
              },
              "ref": null,
              "rendered": "Add task: ",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "onChange": [Function],
              },
              "ref": null,
              "rendered": Array [],
              "type": "select",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Students",
              },
              "ref": null,
              "rendered": "Students",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "className": "studentsArea",
              },
              "ref": null,
              "rendered": Array [],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add task: ",
              },
              "ref": null,
              "rendered": "Add task: ",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "onChange": [Function],
              },
              "ref": null,
              "rendered": Array [],
              "type": "select",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit Course",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Submit Course",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifeCycleMethods": true,
  },
}
`;
